In [30]:
import numpy as np
import pandas as pd

In [31]:
data = pd.read_csv('/content/drive/MyDrive/Tik_Tok_Case_Study/tiktok_dataset.csv')

In [32]:
data

# claim_status    video_id  video_duration_sec  \
0          1        claim  7017666017                  59   
1          2        claim  4014381136                  32   
2          3        claim  9859838091                  31   
3          4        claim  1866847991                  25   
4          5        claim  7105231098                  19   
...      ...          ...         ...                 ...   
19377  19378          NaN  7578226840                  21   
19378  19379          NaN  6079236179                  53   
19379  19380          NaN  2565539685                  10   
19380  19381          NaN  2969178540                  24   
19381  19382          NaN  8132759688                  13   

                                video_transcription_text verified_status  \
0      someone shared with me that drone deliveries a...    not verified   
1      someone shared with me that there are more mic...    not verified   
2      someone shared with me that american industria...    not verified   
3      someone shared with me that the metro of st. p...    not verified   
4      someone shared with me that the number of busi...    not verified   
...                                                  ...             ...   
19377                                                NaN    not verified   
19378                                                NaN    not verified   
19379                                                NaN        verified   
19380                                                NaN    not verified   
19381                                                NaN    not verified   

      author_ban_status  video_view_count  video_like_count  \
0          under review          343296.0           19425.0   
1                active          140877.0           77355.0   
2                active          902185.0           97690.0   
3                active          437506.0          239954.0   
4                active           56167.0           34987.0   
...                 ...               ...               ...   
19377            active               NaN               NaN   
19378            active               NaN               NaN   
19379      under review               NaN               NaN   
19380            active               NaN               NaN   
19381            active               NaN               NaN   

       video_share_count  video_download_count  video_comment_count  
0                  241.0                   1.0                  0.0  
1                19034.0                1161.0                684.0  
2                 2858.0                 833.0                329.0  
3                34812.0                1234.0                584.0  
4                 4110.0                 547.0                152.0  
...                  ...                   ...                  ...  
19377                NaN                   NaN                  NaN  
19378                NaN                   NaN                  NaN  
19379                NaN                   NaN                  NaN  
19380                NaN                   NaN                  NaN  
19381                NaN                   NaN                  NaN  

[19382 rows x 12 columns]

In [33]:
data.head(10)

# claim_status    video_id  video_duration_sec  \
0   1        claim  7017666017                  59   
1   2        claim  4014381136                  32   
2   3        claim  9859838091                  31   
3   4        claim  1866847991                  25   
4   5        claim  7105231098                  19   
5   6        claim  8972200955                  35   
6   7        claim  4958886992                  16   
7   8        claim  2270982263                  41   
8   9        claim  5235769692                  50   
9  10        claim  4660861094                  45   

                            video_transcription_text verified_status  \
0  someone shared with me that drone deliveries a...    not verified   
1  someone shared with me that there are more mic...    not verified   
2  someone shared with me that american industria...    not verified   
3  someone shared with me that the metro of st. p...    not verified   
4  someone shared with me that the number of busi...    not verified   
5  someone shared with me that gross domestic pro...    not verified   
6  someone shared with me that elvis presley has ...    not verified   
7  someone shared with me that the best selling s...    not verified   
8  someone shared with me that about half of the ...    not verified   
9  someone shared with me that it would take a 50...        verified   

  author_ban_status  video_view_count  video_like_count  video_share_count  \
0      under review          343296.0           19425.0              241.0   
1            active          140877.0           77355.0            19034.0   
2            active          902185.0           97690.0             2858.0   
3            active          437506.0          239954.0            34812.0   
4            active           56167.0           34987.0             4110.0   
5      under review          336647.0          175546.0            62303.0   
6            active          750345.0          486192.0           193911.0   
7            active          547532.0            1072.0               50.0   
8            active           24819.0           10160.0             1050.0   
9            active          931587.0          171051.0            67739.0   

   video_download_count  video_comment_count  
0                   1.0                  0.0  
1                1161.0                684.0  
2                 833.0                329.0  
3                1234.0                584.0  
4                 547.0                152.0  
5                4293.0               1857.0  
6                8616.0               5446.0  
7                  22.0                 11.0  
8                  53.0                 27.0  
9                4104.0               2540.0

In [34]:
data.tail(10)

# claim_status    video_id  video_duration_sec  \
19372  19373          NaN  5731766527                  16   
19373  19374          NaN  5731838072                  46   
19374  19375          NaN  3559825127                  42   
19375  19376          NaN  2159797367                  45   
19376  19377          NaN  4099538565                   7   
19377  19378          NaN  7578226840                  21   
19378  19379          NaN  6079236179                  53   
19379  19380          NaN  2565539685                  10   
19380  19381          NaN  2969178540                  24   
19381  19382          NaN  8132759688                  13   

      video_transcription_text verified_status author_ban_status  \
19372                      NaN    not verified            active   
19373                      NaN        verified            active   
19374                      NaN    not verified            active   
19375                      NaN        verified            active   
19376                      NaN    not verified            active   
19377                      NaN    not verified            active   
19378                      NaN    not verified            active   
19379                      NaN        verified      under review   
19380                      NaN    not verified            active   
19381                      NaN    not verified            active   

       video_view_count  video_like_count  video_share_count  \
19372               NaN               NaN                NaN   
19373               NaN               NaN                NaN   
19374               NaN               NaN                NaN   
19375               NaN               NaN                NaN   
19376               NaN               NaN                NaN   
19377               NaN               NaN                NaN   
19378               NaN               NaN                NaN   
19379               NaN               NaN                NaN   
19380               NaN               NaN                NaN   
19381               NaN               NaN                NaN   

       video_download_count  video_comment_count  
19372                   NaN                  NaN  
19373                   NaN                  NaN  
19374                   NaN                  NaN  
19375                   NaN                  NaN  
19376                   NaN                  NaN  
19377                   NaN                  NaN  
19378                   NaN                  NaN  
19379                   NaN                  NaN  
19380                   NaN                  NaN  
19381                   NaN                  NaN

* **The data contains mix of Categorical and Numerical Values.**
* **The data contains few Nan values, which are missing or irrelevant values.**

In [35]:
data.shape

(19382, 12)

In [36]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 19382 entries, 0 to 19381
Data columns (total 12 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   #                         19382 non-null  int64  
 1   claim_status              19084 non-null  object 
 2   video_id                  19382 non-null  int64  
 3   video_duration_sec        19382 non-null  int64  
 4   video_transcription_text  19084 non-null  object 
 5   verified_status           19382 non-null  object 
 6   author_ban_status         19382 non-null  object 
 7   video_view_count          19084 non-null  float64
 8   video_like_count          19084 non-null  float64
 9   video_share_count         19084 non-null  float64
 10  video_download_count      19084 non-null  float64
 11  video_comment_count       19084 non-null  float64
dtypes: float64(5), int64(3), object(4)
memory usage: 1.8+ MB


In [37]:
data.describe()

#      video_id  video_duration_sec  video_view_count  \
count  19382.000000  1.938200e+04        19382.000000      19084.000000   
mean    9691.500000  5.627454e+09           32.421732     254708.558688   
std     5595.245794  2.536440e+09           16.229967     322893.280814   
min        1.000000  1.234959e+09            5.000000         20.000000   
25%     4846.250000  3.430417e+09           18.000000       4942.500000   
50%     9691.500000  5.618664e+09           32.000000       9954.500000   
75%    14536.750000  7.843960e+09           47.000000     504327.000000   
max    19382.000000  9.999873e+09           60.000000     999817.000000   

       video_like_count  video_share_count  video_download_count  \
count      19084.000000       19084.000000          19084.000000   
mean       84304.636030       16735.248323           1049.429627   
std       133420.546814       32036.174350           2004.299894   
min            0.000000           0.000000              0.000000   
25%          810.750000         115.000000              7.000000   
50%         3403.500000         717.000000             46.000000   
75%       125020.000000       18222.000000           1156.250000   
max       657830.000000      256130.000000          14994.000000   

       video_comment_count  
count         19084.000000  
mean            349.312146  
std             799.638865  
min               0.000000  
25%               1.000000  
50%               9.000000  
75%             292.000000  
max            9599.000000

In [38]:
data.head()

# claim_status    video_id  video_duration_sec  \
0  1        claim  7017666017                  59   
1  2        claim  4014381136                  32   
2  3        claim  9859838091                  31   
3  4        claim  1866847991                  25   
4  5        claim  7105231098                  19   

                            video_transcription_text verified_status  \
0  someone shared with me that drone deliveries a...    not verified   
1  someone shared with me that there are more mic...    not verified   
2  someone shared with me that american industria...    not verified   
3  someone shared with me that the metro of st. p...    not verified   
4  someone shared with me that the number of busi...    not verified   

  author_ban_status  video_view_count  video_like_count  video_share_count  \
0      under review          343296.0           19425.0              241.0   
1            active          140877.0           77355.0            19034.0   
2            active          902185.0           97690.0             2858.0   
3            active          437506.0          239954.0            34812.0   
4            active           56167.0           34987.0             4110.0   

   video_download_count  video_comment_count  
0                   1.0                  0.0  
1                1161.0                684.0  
2                 833.0                329.0  
3                1234.0                584.0  
4                 547.0                152.0

In [39]:
data['claim_status'].value_counts()

claim_status
claim      9608
opinion    9476
Name: count, dtype: int64

In [40]:
data_c1 = data[data['claim_status'] == 'claim']
view_count_avg = data_c1['video_view_count'].mean()
view_count_med = data_c1['video_view_count'].median()
print('Video view count Avg for claim status',view_count_avg)
print('Video view count Med for claim status',view_count_med)

Video view count Avg for claim status 501029.4527477102
Video view count Med for claim status 501555.0


In [41]:
data_c1 = data[data['claim_status'] == 'opinion']
view_count_avg = data_c1['video_view_count'].mean()
view_count_med = data_c1['video_view_count'].median()
print('Video view count Avg for opinion status',view_count_avg)
print('Video view count Med for opinion status',view_count_med)

Video view count Avg for opinion status 4956.43224989447
Video view count Med for opinion status 4953.0


In [42]:
data.groupby(['claim_status', 'author_ban_status']).count()[['#']]

#
claim_status author_ban_status      
claim        active             6566
             banned             1439
             under review       1603
opinion      active             8817
             banned              196
             under review        463

In [43]:
data.groupby('author_ban_status').median(numeric_only=True)[['video_share_count']]

video_share_count
author_ban_status                   
active                         437.0
banned                       14468.0
under review                  9444.0

In [44]:
data.groupby('author_ban_status').agg({
    'video_view_count': ['count', 'mean', 'median'],
    'video_like_count': ['count', 'mean', 'median'],
    'video_share_count': ['count', 'mean', 'median']
})

video_view_count                          video_like_count  \
                             count           mean    median            count   
author_ban_status                                                              
active                       15383  215927.039524    8616.0            15383   
banned                        1635  445845.439144  448201.0             1635   
under review                  2066  392204.836399  365245.5             2066   

                                           video_share_count                \
                            mean    median             count          mean   
author_ban_status                                                            
active              71036.533836    2222.0             15383  14111.466164   
banned             153017.236697  105573.0              1635  29998.942508   
under review       128718.050339   71204.5              2066  25774.696999   

                            
                    median  
author_ban_status           
active               437.0  
banned             14468.0  
under review        9444.0

In [45]:
data['likes_per_view'] = data['video_like_count'] / data['video_view_count']
data['comments_per_view'] = data['video_comment_count'] / data['video_view_count']
data['shares_per_view'] = data['video_share_count'] / data['video_view_count']

In [46]:
data.groupby(['claim_status', 'author_ban_status']).agg({
    'likes_per_view': ['count', 'mean', 'median'],
    'comments_per_view': ['count', 'mean', 'median'],
    'shares_per_view': ['count', 'mean', 'median']
})

likes_per_view                      \
                                        count      mean    median   
claim_status author_ban_status                                      
claim        active                      6566  0.329542  0.326538   
             banned                      1439  0.345071  0.358909   
             under review                1603  0.327997  0.320867   
opinion      active                      8817  0.219744  0.218330   
             banned                       196  0.206868  0.198483   
             under review                 463  0.226394  0.228051   

                               comments_per_view                      \
                                           count      mean    median   
claim_status author_ban_status                                         
claim        active                         6566  0.001393  0.000776   
             banned                         1439  0.001377  0.000746   
             under review                   1603  0.001367  0.000789   
opinion      active                         8817  0.000517  0.000252   
             banned                          196  0.000434  0.000193   
             under review                    463  0.000536  0.000293   

                               shares_per_view                      
                                         count      mean    median  
claim_status author_ban_status                                      
claim        active                       6566  0.065456  0.049279  
             banned                       1439  0.067893  0.051606  
             under review                 1603  0.065733  0.049967  
opinion      active                       8817  0.043729  0.032405  
             banned                        196  0.040531  0.030728  
             under review                  463  0.044472  0.035027

# **Insights found**

* The dataframe contains a collection of categorical, text, and numerical data. Each row represents a distinct TikTok video that presents either a claim or an opinion and the accompanying metadata about that video.

* The dataframe contains five float64s, three int64s, and four objects. There are 19,382 observations, but some of the variables are missing values, including claim status, the video transcripton, and all of the count variables.

* Many of the count variables seem to have outliers at the high end of the distribution. They have very large standard deviations and maximum values that are very high compared to their quartile values.

*  The counts of each claim status are quite balanced.

*  The mean and the median within each claim category are close to one another, but there is a vast discrepancy between view counts for videos labeled as claims and videos labeled as opinions.

* There are many more claim videos with banned authors than there are opinion videos with banned authors.

* Banned authors have a median share count that's 33 times the median share count of active authors

* Banned authors and those under review get far more views, likes, and shares than active authors.

* We know that videos by banned authors and those under review tend to get far more views, likes, and shares than videos by non-banned authors. However, when a video does get viewed, its engagement rate is less related to author ban status and more related to its claim status.

* Of the 19,382 samples in this dataset, just under 50% are claims—9,608 of them.

* Engagement level is strongly correlated with claim status.

* Videos with banned authors have significantly higher engagement than videos with active authors

* Videos with authors under review fall between these two categories in terms of engagement levels.


